### PR TITLE
Fixes and new content for Minestuck fluids

### PR DESCRIPTION
**Pull request by Dweblenod**
Whole bundle of changes! 

Most importantly, they all correctly stop fall damage and act more like one would expect for a fluid. This includes changes to movement in the fluid and sound effect changes. Most of the fluids more closely resemble water than lava/default fluids.

Still needs to be done:
- [x] Improve handling of splashing sound effects
- [ ] Potential sound effects for submerging into and emerging from the fluid surface
- [ ] New subtitles for sound effects
- [x] Transparency changes to fluid surface texture
- [x] Fixes to rate at which swimmers sink

### CODE REVIEW

File: src/main/java/com/mraof/minestuck/block/MSBlocks.java [hunk 1]
```diff
@@ -622,12 +622,12 @@ private static Block coalOre(BlockBehaviour.Properties properties)
...
+	public static final RegistryObject<LiquidBlock> BRAIN_JUICE = REGISTER.register("brain_juice", () -> new FlowingModFluidBlock(MSFluids.BRAIN_JUICE, new Vec3(0.55, 0.25, 0.7), 0.25f, Block.Properties.of().mapColor(MapColor.COLOR_PURPLE).replaceable().pushReaction(PushReaction.DESTROY).liquid().noCollission().strength(100.0F).noLootTable().sound(SoundType.EMPTY)));
+	public static final RegistryObject<LiquidBlock> WATER_COLORS = REGISTER.register("water_colors", () -> new FlowingWaterColorsBlock(MSFluids.WATER_COLORS, 0.01f, Block.Properties.of().mapColor(MapColor.WATER).replaceable().pushReaction(PushReaction.DESTROY).liquid().noCollission().strength(100.0F).noLootTable().sound(SoundType.EMPTY)));
+	public static final RegistryObject<LiquidBlock> ENDER = REGISTER.register("ender", () -> new FlowingModFluidBlock(MSFluids.ENDER, new Vec3(0, 0.35, 0.35), (Float.MAX_VALUE), Block.Properties.of().mapColor(MapColor.COLOR_CYAN).replaceable().pushReaction(PushReaction.DESTROY).liquid().noCollission().strength(100.0F).noLootTable().sound(SoundType.EMPTY)));
```
Comment (Dweblenod) [inline src/main/java/com/mraof/minestuck/block/MSBlocks.java:629]: Was there a reason beforehand why the map color was the same as water?

File: src/main/java/com/mraof/minestuck/fluid/MSFluidType.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	static boolean passesMovementThreshold(LivingEntity entity)
+	{
+		Vec3 oldPosVec = new Vec3(entity.xOld, entity.yOld, entity.zOld);
+		Vec3 newPosVec = new Vec3(entity.getX(), entity.getY(), entity.getZ());
+		return oldPosVec.distanceTo(newPosVec) >= 0.05D;
+	}
+	
+	private static void swimSounds(LivingEntity entity, FLUID_STYLE movementType)
+	{
+		SoundEvent swimSound = entity instanceof Player ? SoundEvents.PLAYER_SWIM : SoundEvents.GENERIC_SWIM;
```
Comment (Dweblenod) [inline src/main/java/com/mraof/minestuck/fluid/MSFluidType.java:105]: There are more swim sound effect variants that I ignored

File: src/main/java/com/mraof/minestuck/fluid/MSFluidTypes.java [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+import net.minecraftforge.registries.ForgeRegistries;
+import net.minecraftforge.registries.RegistryObject;
+
+import java.util.function.Consumer;
+
+public class MSFluidTypes
+{
+	public static final DeferredRegister<FluidType> REGISTER = DeferredRegister.create(ForgeRegistries.Keys.FLUID_TYPES, Minestuck.MOD_ID);
+	
+	//TODO fix flowing fluid push mechanics (player is too slow to walk up stream)
```
Comment (Dweblenod) [inline src/main/java/com/mraof/minestuck/fluid/MSFluidTypes.java:20]: Not sure where would be best to change this to accommodate the problem

File: src/main/java/com/mraof/minestuck/fluid/MSFluidTypes.java [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+				{
+					return FLOWING_TEXTURE;
+				}
+			});
+		}
+	});
+	
+	public static final RegistryObject<FluidType> BRAIN_JUICE_TYPE = REGISTER.register("brain_juice", () -> new MSFluidType(FluidType.Properties.create()
+			.canHydrate(true)
+			.canDrown(false) //imagining it as a fantastical breathable liquid
```
Comment (Dweblenod) [inline src/main/java/com/mraof/minestuck/fluid/MSFluidTypes.java:140]: Any feelings on this change?

File: src/main/java/com/mraof/minestuck/block/MSBlocks.java (hunk 1 above)
Comment (kirderf1) [reply src/main/java/com/mraof/minestuck/block/MSBlocks.java:629]: Likely only as a copy-paste of water block properties/water material color. Changing map color should be fine as far as I'm concerned.

File: src/main/java/com/mraof/minestuck/fluid/MSFluidTypes.java (hunk 3 above)
Comment (Dweblenod) [reply src/main/java/com/mraof/minestuck/fluid/MSFluidTypes.java:20]: Still unsure how to fix this without also resulting in changes to movement in source blocks

**Pull request merged**